test/indexed_test.dart

```diff
@@ -0,0 +1,199 @@
+
+Future<void> main([List<String>? args]) async {
+  test('Indexed faster', () {
+    final config = Configuration.local([WithIndexes.schema, NoIndexes.schema]);
+    Realm.deleteRealm(config.path);
+    final realm = Realm(config);
+
+    const max = 100000;
+    final allIndexed = realm.all<WithIndexes>();
+    final allNotIndexed = realm.all<NoIndexes>();
+
+    expect(realm.all<WithIndexes>().length, 0);
+
+    intFactory(int i) => i.hashCode;
+    boolFactory(int i) => i % 2 == 0;
+    stringFactory(int i) => '${i.hashCode} $i';
+    timestampFactory(int i) => DateTime.fromMillisecondsSinceEpoch(i.hashCode);
+    objectIdFactory(int i) => ObjectId.fromValues(i.hashCode * 1000000, i.hashCode, i);
+    uuidFactory(int i) => Uuid.fromBytes(Uint8List(16).buffer..asByteData().setInt64(0, i.hashCode));
+
+    final indexed = List.generate(
+      max,
+      (i) => WithIndexes(
+        intFactory(i),
+        boolFactory(i),
+        stringFactory(i),
+        timestampFactory(i),
+        objectIdFactory(i),
+        uuidFactory(i),
+      ),
+    );
+    realm.write(() => realm.addAll(indexed));
+
+    expect(allIndexed.length, max);
+
+    final notIndexed = List.generate(
+      max,
+      (i) => NoIndexes(
+        intFactory(i),
+        boolFactory(i),
+        stringFactory(i),
+        timestampFactory(i),
+        objectIdFactory(i),
+        uuidFactory(i),
+      ),
+    );
+    realm.write(() => realm.addAll(notIndexed));
+
+    expect(allNotIndexed.length, max);
+
+    // Inefficient, but fast enough for this test
+    final searchOrder = (List.generate(max, (i) => i)..shuffle(Random(42))).take(1000).toList();
+
+    QueryBuilder<T, U> builder<T extends RealmObject, U extends Object>(RealmResults<T> results, String fieldName) {
+      return (U value) => results.query('$fieldName == \$0', [value]);
+    }
+
+    @pragma('vm:no-interrupts')
+    Duration measureSpeed<T extends RealmObject, U extends Object>(
+      RealmResults<T> results,
+      String fieldName,
+      U Function(int index) indexToValue,
+    ) {
+      final queryBuilder = builder(results, fieldName);
+      final queries = searchOrder.map((i) => queryBuilder(indexToValue(i))).toList(); // pre-calculate queries
+      final found = <T?>[];
+
+      final sw = Stopwatch()..start();
+      for (final q in queries) {
+        found.add(q.singleOrNull); // evaluate query
+      }
+      final timing = sw.elapsed;
+
+      // check that we found the right objects
+      for (final f in found) {
+        expect(f, isNotNull);
+      }
+
+      return timing;
+    }
+
+    void compareSpeed<U extends Object>(
+      String fieldName,
+      U Function(int index) indexToValue,
+    ) {
+      final lookupCount = searchOrder.length;
+
+      display(Type type, Duration duration) {
+        print('$lookupCount lookups of ${'$type'.padRight(12)} on ${fieldName.padRight(10)} : ${duration.inMicroseconds ~/ lookupCount} us/lookup');
```

Comment:
Do we need to print every time. Why not do that only if something is off.

Reply:
No we don't need to

Reply:
Changed it so measurements are only displayed on test failure